spark/src/main/java/org/opensearch/sql/spark/execution/session/CreateSessionRequest.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.execution.session;
+
+import lombok.Data;
+import org.opensearch.sql.spark.client.StartJobRequest;
+
+@Data
+public class CreateSessionRequest {
+  private final StartJobRequest startJobRequest;
+  private final String datasourceName;
```

Comment:
Do we require datasourceName in asyncQuery Request model?

Reply:
datasourceName is write to Session doc in flint.ql.sessions index. it is required for user perform doc level access control.

Reply:
I am actually asking for this api _plugins/_async_query. I think we need it. 
Should we perform datasource authorization in both the calls?

Reply:
1. for create session, _plugins/_async_query/sessions, require datasource, address it in separate PR. 
2. we need datasource authZ for both _plugins/_async_query/sessions and _plugins/_async_query API.